### ISSUE TITLE
Broken lints sugesstion on `format!` macro

### ISSUE THREAD

**Issue by lzutao** [rust-lang/rust-clippy#4309]
Maybe related to #3630.

### Example

```rust
fn foo(fail: bool) -> Result<i32, String> {
    if fail {
      Err(format!("failed {:?}", fail))?;
    }
    Ok(0)
}
```

### Expected Behavior

```rust
warning: returning an `Err(_)` with the `?` operator
 --> src/lib.rs:3:7
  |
3 |       Err(format!("failed {:?}", fail))?;
  |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `return Err(format!("failed {:?}", fail))`
  |
  = note: `#[warn(clippy::try_err)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#try_err
```

### Current Behavior

```rust
warning: returning an `Err(_)` with the `?` operator
 --> src/lib.rs:3:7
  |
3 |       Err(format!("failed {:?}", fail))?;
  |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `return Err($ crate :: fmt :: format (format_args ! ($ ($ arg) *)))`
  |
  = note: `#[warn(clippy::try_err)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#try_err
```

### `cargo clippy -V`
clippy 0.0.212 (dc69a5c 2019-07-27)

**Comment by flip1995** [comment]
`good first issue`: Suppressing the macro expansion on the creation of the suggestion was already done a few times in Clippy.

**Comment by oli-obk** [comment]
In case we don't already have a helper function in utils, this would be a good time to pull one out. Maybe just make `snippet` and `snippet_opt` resolve all macro expansions first?
